Backend/AlgorithmsManager.py

```diff
@@ -513,36 +492,26 @@ def sort_tweets(arr):
 
             arr[k + 1] = temp
 
-
+    # pass in a dictionary and the length of the returned shortlist
+    # return a list of tuples, sorted by the retweet count
     @staticmethod
-    def sort_retweeters(arr):
-        length = len(arr)
-        for l in range(1, length):
-            temp = arr[l]
-            tempval = temp['retweets_count']
-            k = l-1
-
-            num_trend_sort_iter = 0
-
-            while k >= 0 and    (tempval if tempval is not None else 0) >= \
-                                (arr[k]['retweets_count'] if arr[k]['retweets_count'] is not None else 0):
-                arr[k+1] = arr[k]
-                k -= 1
-
-                # todo strip debugging code from while loop
-                num_trend_sort_iter +=1
-                if num_trend_sort_iter >= 10000:
-                    print("while loop started to go infinite:\nARGUMENTS:\n",
-                          "arr= ", arr,
-                          ";\n\nVARIABLES:\n",
-                          "temp= ", temp, "\ntempval= ", tempval, "\nk= ", k,
-                          "\nSTACK TRACE:\n")
-
-                    traceback.print_stack()
-                    raise Exception("got into infinite loop")
-
-            arr[k+1] = temp
-
+    def sort_retweeters(dict, count):
+        sorted_list = []
+        keyList = list(dict)
+
+        for k in range(count):
+            temp_key = keyList[0]
+            temp_val = dict[temp_key]
+            for iter_key in keyList:
+                if dict[iter_key] >= temp_val:
+                    temp_key = iter_key
+                    temp_val = dict[iter_key]
+            
+            user = {"username": temp_key, "retweet_count": temp_val}
+            sorted_list.append(user)
+            dict[temp_key] = -1
```

Comment:
This looks solid. Just as a note for later, if we find that this is a peformance bottleneck (which we wouldn't know until we tried profiling all of the whole endpoint's code), I think there's an at least somewhat faster implementation which might be worth the nonnegligible trouble IF we found that this was holding things up.
We can rewrite this to loop through the dict's keys only once by building a list of length 'count' containing the keys which had the highest values among the keys looked at so far.

Reply:
sounds good!